### DISCUSSION TITLE
Helmfile runner for java

### DISCUSSION THREAD

**Discussion by paichinger** [helmfile/helmfile#995] [Show and tell]
Hey folks!
At my job (Platform Engineer @ [topdesk](https://www.topdesk.com/)) we're using helmfile extensively. We have a very large helmfile setup we're using to deploy all our stuff to production, acceptance and testing environments and even developer machines.  
  
One thing we we're always struggling with was the testability of helmfile. We want to know if a change to our templates would break stuff! While we have extensive tests on all levels of our deployment pipelines, we didn't have any way of testing our quite complex templating we're doing in helmfile. As we couldn't find anything out there I decided to create our own solution. TOPdesk is a java shop, so we have a lot of developers with profound java skills. Also, the junit framework is most likely the most mature unit test framework out there. That's why I decided to create a java runner for helmfile -> [jhelmfile](https://github.com/paichinger/jhelmfile), so we can use it to run unit tests against our helmfile setup. However, jhelmfile can also be used to run helmfile in any other scenario not just for unit tests. 
   
I want to introduce this project here, because I think it fills a gap and might be useful for other people as well. It's far from feature parity with the helmfile cli but I'm happy to add any missing command/parameter if needed.  
I was also hoping that we could maybe add a section for testing in the helmfile documentation and put a short chapter about jhelmfile there to get more attention from people who want to write tests for helmfile and struggle with the same problem we did.  
  
Let me know if this is interesing for you!
Regards,
Peter

**Comment by yxxhero** [comment]
@paichinger What can I do for you? PR is welcome. it's an useful tools.

**Reply by paichinger** [reply]
Thanks, but for now I think I'm OK on my own. I'll come up with a PR for the docs in the near future. Just wanted to introduce and verify the issue with you guys before I do so.

**Reply by yxxhero** [reply]
@paichinger free to do that you want to do.
